src/components/controls/buttons/DropdownButton.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from '../../controls/buttons/Button';
+import RootCloseWrapper from 'react-overlays/lib/RootCloseWrapper';
+import styled, { ThemeProvider } from 'styled-components';
+import defaultTheme from '../../theme/defaultTheme';
+
+const Caret = styled.span`
+  display: inline-block;
+  width: 0;
+  height: 0;
+  margin-left: 5px;
+  vertical-align: middle;
+  border-top: 4px dashed;
+  border-right: 4px solid transparent;
+  border-left: 4px solid transparent;
+`;
+
+const ButtonPanel = styled.div`
+  display: ${props => (props.isOpen ? 'block' : 'none')};
+  margin-top: 3px;
+  position: absolute;
+  border: 1px solid ${props => props.theme.colors.grayLight};
+  background-color: ${props => props.theme.colors.white};
+`;
+
+const ButtonPanelChildrenContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const StyledButtonLink = styled(Button)`
+  padding: 10px 20px;
+  color: black;
+  text-decoration: none;
+  text-align: left;
+  margin-bottom: 0px;
+  &:hover {
+    background-color: ${props => props.theme.colors.grayLighter};
+  }
+`;
+
+class DropdownButton extends React.Component {
```

Comment:
You can't guarantee that all child components will be pure. The performance boost is likely negligible. I wouldn't introduce it unless we've profiled the component itself and found a performance problem in rendering.

Reply:
I didn't realize that pure components require pure children. That does seem a little odd.

Reply:
It's because `PureComponent` will skip prop updates for the entire subtree. I think if you go with the `<Dropdown.Button>` idea, you could make that pure and then use `PureComponent`. Again, the performance gains are likely negligible.